doc/dev: Fix broken search links in `Life of a Search Query`

indradhanush opened the pull request:
This PR fixes search links for the article: Life of a Search Query

### Search query for `schema.graphql`

#### Current behaviour:
![image](https://user-images.githubusercontent.com/2682729/111964652-da21f080-8b1a-11eb-9b63-9822ffc4a6c1.png)

#### Expected behaviour with this PR:
![image](https://user-images.githubusercontent.com/2682729/112105502-807cfd00-8bd2-11eb-9179-1edd19dde3d1.png)


### Search query for the GraphQL search resolver

#### Current behaviour:

![image](https://user-images.githubusercontent.com/2682729/111969526-5a972000-8b20-11eb-9f6f-c0435ae30f57.png)

#### Expected behaviour with this PR:

![image](https://user-images.githubusercontent.com/2682729/111969470-45ba8c80-8b20-11eb-80aa-488e4449d940.png)

### Search query for searchResolver.resolveRepositories

#### Current behaviour:

![image](https://user-images.githubusercontent.com/2682729/111977141-6dadee00-8b28-11eb-8eac-433cf08e4fb7.png)

#### Expected behaviour with this PR:

![image](https://user-images.githubusercontent.com/2682729/111977070-57079700-8b28-11eb-914f-8f70f211c0d7.png)
 
 ### Search query for zoektIndexedRepos
 
 #### Current behaviour: 
 
![image](https://user-images.githubusercontent.com/2682729/112099844-3d1e9080-8bca-11eb-9a5f-be1580c19caf.png)

#### Expected behaviour with this PR:

![image](https://user-images.githubusercontent.com/2682729/112101712-1d3c9c00-8bcd-11eb-8e26-daba592ee08f.png)


<!-- Reminder: Have you updated the changelog and relevant docs (user docs, architecture diagram, etc) ? -->

### Search query for `Zoekt.start`:

#### Current behaviour:

![image](https://user-images.githubusercontent.com/2682729/112101916-786e8e80-8bcd-11eb-94e6-b30eb6898518.png)

#### Expected behaviour with this PR:

![image](https://user-images.githubusercontent.com/2682729/112102008-9c31d480-8bcd-11eb-9fef-cdc4a1951330.png)

### Indexed repo search with zoekt:

#### Current behaviour:

![image](https://user-images.githubusercontent.com/2682729/112102177-e024d980-8bcd-11eb-8407-be1c14c79b8c.png)

#### Expected behaviour with this PR:

![image](https://user-images.githubusercontent.com/2682729/112265951-5f331400-8c99-11eb-854e-fe638421bfd0.png)


### Non indexed repo search:

#### Current behaviour:
![image](https://user-images.githubusercontent.com/2682729/112102355-267a3880-8bce-11eb-8749-7ae23aa70ae4.png)

#### Expected behaviour:
![image](https://user-images.githubusercontent.com/2682729/112102880-eebfc080-8bce-11eb-85c2-548cf009700a.png)

rvantonder commented on doc/dev/background-information/architecture/life-of-a-search-query.md:
```diff
@@ -10,13 +10,13 @@ There are a few ways to perform a search with Sourcegraph:
 3. Using the [src CLI command](https://github.com/sourcegraph/src-cli).
 
-In all cases, clients use the [search query](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+%5Cbsearch%5C%28+file:schema.graphql) in our GraphQL API that is exposed by our [frontend](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/tree/cmd/frontend) service.
+In all cases, clients use the [search query](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:schema.graphql&patternType=literal) in our GraphQL API that is exposed by our [frontend](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/tree/cmd/frontend) service.
```
```suggestion
In all cases, clients use the [search query](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+%5Cbsearch%5C%28+file:schema.graphql&patternType=regexp) in our GraphQL API that is exposed by our [frontend](https://sourcegraph.com/github.com/sourcegraph/sourcegraph/-/tree/cmd/frontend) service.
```

This is supposed to be with regexp toggled on. Looks like this doc hasn't been updated since we introduced literal search (a long time).

rvantonder replied:
[linke to query](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+%5Cbsearch%5C%28+file:schema.graphql&patternType=regexp)

rvantonder reviewed (approved):
Approving module the one suggested change--if this is still WIP, feel free to conver it to draft and let me know to take another look.

indradhanush commented on doc/dev/background-information/architecture/life-of-a-search-query.md:
```diff
@@ -18,7 +18,7 @@ The frontend implements the GraphQL search resolver [here](https://sourcegraph.c
 First, the frontend [resolves which repositories need to be searched](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+func+%28r+*searchResolver%29+resolveRepositories%28&patternType=literal). It parses the query for any repository filters and then queries the database for the list of repositories that match those filters. If no filters are provided then all repos ...
 
-Next, the frontend [determines which repository@revision combinations are indexed by zoekt](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+zoektIndexedRepos%28+file:zoekt.go&patternType=literal) by [consulting an in-memory cache that is kept up-to-date with regular asynchronous polling](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegra ...
+Next, the frontend [determines which repository@revision combinations are indexed by zoekt](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+zoektIndexedRepos%28+file:zoekt.go&patternType=literal) by [consulting an in-memory cache that is kept up-to-date with regular asynchronous polling](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegra ...
```
This is a change in the actual search term. I was not able to find any function by the name of `zoektSearchHead` in the codebase (although there's a reference to it as a string in a `t.Error` function). I dug deeper down the :rabbit: :hole: and replaced the search with what I thought should be the search term instead. 

So I would like to draw special attention to this change for review to verify if my change is correct or not. :smile:

rvantonder replied:
Yeah the code's changed. I would use [this](https://sourcegraph.com/search?q=repo:%5Egithub%5C.com/sourcegraph/sourcegraph%24+file:%5Ecmd/frontend/graphqlbackend/textsearch%5C.go+indexed.Search%28&patternType=literal) for "It concurrently queries zoekt".

indradhanush commented:
@rvantonder Thank you for your earlier review. I have incorporated your suggestions and also added some new changes since then. While I believe there are still some more broken links in that article, but in the interest of keeping this PR concise and easier to review I am stopping here. I will create a new PR with a second set of changes in the second half of the article. :smile:
